android/src/main/java/org/mozilla/mozstumbler/client/Updater.java

```diff
@@ -40,6 +42,12 @@ public boolean wifiExclusiveAndUnavailable(Context c) {
         return !NetworkInfo.getInstance().isWifiAvailable() && ClientPrefs.getInstance(c).getUseWifiOnly();
     }
 
+    public boolean checkForUpdatesRateLimited(final Activity activity, String api_key) {
+        if (System.currentTimeMillis() - sLastUpdateCheck < UPDATE_CHECK_FREQ) {
+            return false;
+        }
+        return this.checkForUpdates(activity, api_key);
```

Comment:
Let's just inline the `checkForUpdates` method call or at least turn it private as there's no reason to have two publicly accessible entry points into this code.